crates/lakekeeper/migrations/20251128101923_enable_tasks_on_project_level.sql

```diff
@@ -0,0 +1,96 @@
+-- Adjust task tables to enable tasks on project level.
+-- * task
+-- * task_config
+-- * task_log
+
+-- 1. Add project_id columns to all tables (nullable initially)
+alter table task
+add column if not exists project_id text null references project (project_id) on delete cascade;
+
+alter table task_log
+add column if not exists project_id text null references project (project_id) on delete cascade;
+
+alter table task_config
+add column if not exists project_id text null references project (project_id) on delete cascade;
+
+-- 2. Fill project_id for all tables
+with project_info as (
+	select project_id, warehouse_id
+	from warehouse
+),
+task_updates as (
+	update task
+	set project_id = p.project_id
+	from project_info as p
+	where task.project_id is null
+		and task.warehouse_id = p.warehouse_id
+	returning 1
+),
+task_config_updates as (
+	update task_config
+	set project_id = p.project_id
+	from project_info as p
+	where task_config.project_id is null
+		and task_config.warehouse_id = p.warehouse_id
+	returning 1
+)
+update task_log
+set project_id = p.project_id
+from project_info as p
+where task_log.project_id is null
+	and task_log.warehouse_id = p.warehouse_id;
+
+-- 3. Set NOT NULL constraints on project_id
+alter table task
+alter column project_id set not null;
+
+alter table task_config
+alter column project_id set not null;
+
+alter table task_log
+alter column project_id set not null;
+
+-- 4. Modify task_config: drop old PK and add new column + PK
+alter table task_config 
+	drop constraint if exists task_config_pkey,
+	add column if not exists task_config_id uuid default gen_random_uuid(),
+	add primary key (task_config_id);
+
+-- 5. Modify task constraints and make warehouse_id optional
+alter table task 
+	drop constraint if exists task_unique_warehouse_id_entity_type_entity_id_queue_name,
+	alter column warehouse_id drop not null,
+	alter column entity_id drop not null,
+	alter column entity_name drop not null,
+	add unique nulls not distinct (project_id, warehouse_id, entity_type, entity_id, queue_name),
```

Comment:
For more complex constraint we should specify names. This has two advantages:
- Executing it twice fails (currently it does not, it would just get a new name)
- We don't rely on consisten postgres name generation if we ever want to change this constraint.